invenio_users_resources/permissions.py

```diff
@@ -1,13 +1,75 @@
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2023 CERN.
+# Copyright (C) 2025 Ubiquity Press.
 #
 # Invenio-Users-Resources is free software; you can redistribute it and/or modify
 # it under the terms of the MIT License; see LICENSE file for more details.
 """Users resources generic needs and permissions."""
 
-from invenio_access import action_factory
+from invenio_access import ActionRoles, Permission, action_factory, superuser_access
+from invenio_records_permissions.generators import AdminAction
+from invenio_search.engine import dsl
 
 USER_MANAGEMENT_ACTION_NAME = "administration-moderation"
 
 user_management_action = action_factory(USER_MANAGEMENT_ACTION_NAME)
+
+
+class SuperUserMixin:
+    """Mixin for superuser permissions."""
+
+    def _is_group_superadmin(self, group_id, roles):
+        groups = {role.role.id for role in roles}
+        if group_id in groups:
+            return True
+        return False
+
+    def _get_superadmin_roles(self):
```

Comment:
Does it make sense to factor this one out and [cache](https://docs.python.org/3/library/functools.html#functools.cache) it?
You'll probably use it within the same request inside `AdministratorAction`.